### ISSUE TITLE
Hide sync mailbox via imap-acl?

### ISSUE THREAD

**Issue by Valodim** [autocrypt/autocrypt#11]
More of an optional thing:

There is an ACL extension for imap in [rfc 4314](https://www.ietf.org/rfc/rfc4314.txt), which seems to be supported [fairly widely](http://www.imapwiki.org/Specs) (though not by gmail).

If a server supports this extension, a user can do `SETACL $mailbox $username -l` to remove their own `l` permission for that mailbox, effectively hiding the mailbox from listing commands (`LIST`, `LSUB`, `SUBSCRIBE`). This does not prevent access (`SELECT`), so a client that knows about the mailbox won't be bothered.

At first glance this allows us to avoid spamming a user's folder list with "non-interactive", and can be done optionally with no direct cost. There *is* an indirect cost, since it also makes manual intervention ("hackability") much harder.

**Comment by dkg** [comment]
I like this suggestion as something that the clients "MAY" do when using an IMAP folder for sync/lockout

**Comment by hpk42** [comment]
makes sense to me. @valodim can you do a PR that does this "may" and includes the references you already listed?

**Comment by Valodim** [comment]
I hacked a bit on this. In general, removing the "l" permission works fine to hide the folder from view, while keeping it accessible for a SELECT. I only tested with dovecot, can't say about other imap implementations.

One finding I had was that dovecot had a bug with differential ACLs if none exist yet :) I [reported the bug](http://dovecot.org/pipermail/dovecot/2017-January/106592.html) but didn't get an answer :(
